File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/UserEntityMapper.java
package br.org.restaurantepopular.data_db.mapper;

import br.org.restaurantepopular.data_db.entity.UserEntity;
import br.org.restaurantepopular.entity.User;

public class UserEntityMapper {
    public static UserEntity toUserEntity(User user){
        return new UserEntity.Builder()
                .id(user.getId())
                .name(user.getName())
                .role(user.getRole())
                .registration(user.getRegistration())
                .cpf(user.getCpf())
                .build();
    }

    public static User toUser(UserEntity userEntity) {
        return new User.UserBuilder()
                .id(userEntity.getId())
                .name(userEntity.getName())
                .role(userEntity.getRole())
                .registration(userEntity.getRegistration())
                .cpf(userEntity.getCpf())
                .build();
    }
}

File: data_db/src/main/java/br/org/restaurantepopular/data_db/mapper/UserEntityMapper.kt
package br.org.restaurantepopular.data_db.mapper

import br.org.restaurantepopular.data_db.entity.UserEntity
import br.org.restaurantepopular.entity.User

object UserEntityMapper {
    fun toUserEntity(user: User): UserEntity {
        return UserEntity(
            id = user.id ?: "",
            name = user.name,
            role = user.role,
            code = user.code,
            cpf = user.cpf
        )
    }

    fun toUser(userEntity: UserEntity): User {
        return User(
            id = userEntity.id,
            name = userEntity.name,
            role = userEntity.role,
            code = userEntity.code,
            cpf = userEntity.cpf
        )
    }
}